### PR TITLE
Add bitmovin

### PR DESCRIPTION
**Pull request by jamescyeh**
Adding capability to submit jobs and transcode using the Bitmovin Cloud Encoding Service.

### CODE REVIEW

File: service/transcode.go [hunk 1]
```diff
@@ -84,6 +84,8 @@ func (s *TranscodingService) newTranscodeJob(r *http.Request) swagger.GizmoJSONR
 	if err != nil {
 		return swagger.NewErrorResponse(err)
 	}
+	fmt.Println("should be 200'ing here")
+	fmt.Println(job.ID)
```
Comment (fsouza) [inline service/transcode.go]: You can delete this too! :D

File: provider/bitmovin/bitmovin.go [hunk 2]
```diff
@@ -0,0 +1,779 @@
...
+	if preset.Video.Bitrate == "" {
+		return nil, errors.New("Video Bitrate must be set")
+	}
+	bitrate, err := strconv.Atoi(preset.Video.Bitrate)
+	if err != nil {
+		return nil, err
+	}
+	h264.Bitrate = intToPtr(int64(bitrate))
+	if preset.Video.GopSize != "" {
+		gopSize, err := strconv.Atoi(preset.Video.GopSize)
```
Comment (fsouza) [inline provider/bitmovin/bitmovin.go]: ugh, sorry for making you do that

File: provider/bitmovin/bitmovin.go [hunk 3]
```diff
@@ -0,0 +1,779 @@
...
+			}
+			if audioMediaInfoResp.Status == bitmovinAPIErrorMsg {
+				return nil, errors.New("Error in adding EXT-X-MEDIA")
+			}
+
+			// create the video ts muxing
+
+			// videoMuxingStream := models.StreamItem{
+			// 	StreamID: &videoStreamID,
+			// }
```
Comment (fsouza) [inline provider/bitmovin/bitmovin.go]: Can you delete these commented lines?

File: provider/bitmovin/bitmovin.go (hunk 2 above)
Comment (jamescyeh) [inline provider/bitmovin/bitmovin.go]: No worries!

File: service/transcode.go (hunk 1 above)
Comment (jamescyeh) [inline service/transcode.go]: Done

File: provider/bitmovin/bitmovin.go (hunk 3 above)
Comment (jamescyeh) [inline provider/bitmovin/bitmovin.go]: Done.

**Comment by fsouza** [comment]
Thank you very much for working on that!

**Pull request merged**